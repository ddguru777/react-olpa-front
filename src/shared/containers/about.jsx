/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the GPL v2.0+ license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { appSetTitle } from "zoapp-front/actions/app";

const aboutStyle = {
  fontSize: "18px",
  fontWeight: "300",
  color: "#666",
  margin: "36px",
};
const infoStyle = {
  fontSize: "24px",
  fontWeight: "300",
  color: "#666",
  paddingBottom: "20px",
};

export class About extends Component {
  componentWillMount() {
    this.props.appSetTitle("About");
  }

  render() {
    const { isLoading } = this.props;

    return (
      <div style={aboutStyle}>
        <div style={infoStyle}>{isLoading ? "" : null} </div>
        <p>
          We are Opla, the mightly volcanic team ! Since 2015 we started
          developing this deeptech chatbot solution. We are truly commited to
          OpenSource. And we are fully involved in creating the best and
          complete tool for conversational interface, user oriented.
        </p>
        <p>
          Version<br />0.1 : Amazing Alou (Le Puy Alou : a french volcano) June
          2018<br />
          <a
            href="https://github.com/Opla/"
            target="_blank"
            rel="noreferrer noopener"
          >
            See more on Github
          </a>
        </p>
        <p>
          Thanks to the team and contributors<br />
          Julie, Maali, Marie, Nicolle, Dorian, Loic, Manu, Mik, Thomas, Will,
          Will
        </p>
        <p>
          Follow us on Twitter @opla_ai<br />
          Or contact us hello@opla.ai
        </p>
      </div>
    );
  }
}

About.propTypes = {
  isLoading: PropTypes.bool,
  appSetTitle: PropTypes.func.isRequired,
};
const mapDispatchToProps = (dispatch) => ({
  appSetTitle: (titleName) => {
    dispatch(appSetTitle(titleName));
  },
});

export default connect(null, mapDispatchToProps)(About);
