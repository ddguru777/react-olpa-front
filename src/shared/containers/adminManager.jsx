/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the GPL v2.0+ license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React, { Component } from "react";
import PropTypes from "prop-types";
import Zrmc, {
  Grid,
  Inner,
  Cell,
  Button,
  Icon,
  TextField,
  Select,
  MenuItem,
} from "zrmc";
import { TableComponent } from "zoapp-ui";
import { connect } from "react-redux";
import Loading from "zoapp-front/components/loading";
import SignInForm from "zoapp-front/containers/signInForm";
import { apiSetAdminParametersRequest } from "zoapp-front/actions/api";

import { appSetTitle } from "../actions/app";
import { apiSaveBotRequest } from "../actions/api";
import ServicesContainer from "./servicesContainer";
import PluginsManager from "../utils/pluginsManager";
import TunnelBox from "../components/tunnelBox";
import timezones from "../utils/timezones";

const infoStyleD = {
  fontSize: "16px",
  fontWeight: "400",
  color: "#666",
  padding: "24px",
  lineHeight: "1.1",
};

const FORM_WIDTH = "100%";

class AdminManager extends Component {
  static onActionTunnel(/* dialog, action */) {
    Zrmc.closeDialog();
  }

  constructor(props) {
    super(props);
    const pluginsManager = new PluginsManager();

    this.state = {
      pluginsManager,
      bot: props.bot,
      tunnelParams: null,
      backendParams: null,
      emailServerParams: null,
    };
  }

  componentWillMount() {
    this.props.appSetTitle("Admin");
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.bot !== this.props.bot) {
      this.setState({
        bot: nextProps.bot,
      });
    }
  }

  onChangeTunnel = (tunnelParams) => {
    this.setState({ tunnelParams });
  };

  onSaveBotDetails = () => {
    this.props.apiSaveBotRequest(this.state.bot);
  };

  onSaveBackend() {
    if (this.state.tunnelParams) {
      // WIP save tunnel parameters
      const tunnel = this.state.tunnelParams.active || "None";
      this.setState({ tunnelParams: null });
      this.props.apiSetAdminParametersRequest({ tunnel });
    }
  }

  displayTunnelDialog() {
    const { params } = this.props.admin;
    const backend = this.state.backendParams || params.backend || {};
    const tunnelParams = this.state.tunnelParams || backend.tunnel || {};
    const body = (
      <TunnelBox onChange={this.onChangeTunnel} params={tunnelParams} />
    );
    Zrmc.showDialog({
      header: "Tunnel settings",
      body,
      syle: { width: "520px" },
      onAction: AdminManager.onActionTunnel,
    });
  }

  handleBotNameChange = (e) => {
    const name = e.target.value;

    this.setState({
      bot: {
        ...this.state.bot,
        name,
      },
    });
  };

  handleBotDescriptionChange = (e) => {
    const description = e.target.value;

    this.setState({
      bot: {
        ...this.state.bot,
        description,
      },
    });
  };

  handleLanguageChange = (language) => {
    this.setState({
      bot: {
        ...this.state.bot,
        language,
      },
    });
  };

  handleTimezoneChange = (timezone) => {
    this.setState({
      bot: {
        ...this.state.bot,
        timezone,
      },
    });
  };

  render() {
    let { isLoading } = this.props;
    if (!isLoading && !this.props.admin && this.props.isSignedIn) {
      isLoading = true;
    }
    if (!this.props.isSignedIn) {
      return <SignInForm />;
    } else if (isLoading || this.props.admin == null) {
      return <Loading />;
    }
    const active = this.props.activeTab;
    let content = "";
    if (active === 0) {
      content = (
        <Grid>
          <Inner>
            <Cell
              className="mdl-color--white"
              span={12}
              style={{ display: "table" }}
            >
              <div style={{ width: "200px", display: "table-cell" }}>
                <div
                  style={{
                    position: "absolute",
                    width: "180px",
                    height: "180px",
                    margin: "24px",
                    backgroundColor: "#ddd",
                  }}
                />
              </div>
              <div style={{ display: "table-cell" }}>
                <form style={infoStyleD}>
                  <div>
                    <TextField
                      defaultValue={this.props.bot.name}
                      label="Assistant name"
                      onChange={this.handleBotNameChange}
                      style={{ width: FORM_WIDTH }}
                    />
                  </div>
                  <div>
                    <TextField
                      defaultValue={this.props.bot.description}
                      isTextarea
                      label="Description"
                      onChange={this.handleBotDescriptionChange}
                      style={{ width: FORM_WIDTH }}
                    />
                  </div>
                  <div>
                    <Select
                      label="Language"
                      onSelected={this.handleLanguageChange}
                      style={{ width: FORM_WIDTH }}
                      selectedIndex={["en", "fr"].findIndex(
                        (language) =>
                          language === (this.props.bot.language || null),
                      )}
                    >
                      <MenuItem value="en">English</MenuItem>
                      <MenuItem value="fr">French</MenuItem>
                    </Select>
                  </div>
                  <div>
                    <Select
                      label="Timezone"
                      onSelected={this.handleTimezoneChange}
                      style={{ width: FORM_WIDTH }}
                      selectedIndex={this.props.timezones.findIndex(
                        (tz) => tz === (this.props.bot.timezone || null),
                      )}
                    >
                      {this.props.timezones.map((timezone) => (
                        <MenuItem key={timezone} value={timezone}>
                          {timezone}
                        </MenuItem>
                      ))}
                    </Select>
                  </div>
                </form>
              </div>
              <div>
                <Button
                  disabled={
                    !this.state.bot ||
                    (this.state.bot && this.state.bot.name.length < 1)
                  }
                  raised
                  style={{ float: "right", margin: "24px" }}
                  onClick={this.onSaveBotDetails}
                >
                  SAVE
                </Button>
              </div>
            </Cell>
          </Inner>
        </Grid>
      );
    } else if (active === 1) {
      content = (
        <Grid>
          <ServicesContainer pluginsManager={this.state.pluginsManager} />
        </Grid>
      );
    } else if (active === 2) {
      const items = [];
      const status = "you";
      const { user, profile } = this.props;

      const values = [];
      values.push(profile.username);
      values.push(profile.email);
      values.push(user.attributes.scope);
      values.push(status);

      items.push({ id: 1, values, icon: `../images/${profile.avatar}.png` });

      const headers = ["", "username", "email", "role", "status"];
      const title = (
        <div style={infoStyleD}>
          You could give an access to your collaborators here.
          <Button raised style={{ float: "right", marginBottom: "24px" }}>
            ADD
          </Button>
        </div>
      );

      content = (
        <Grid>
          <Inner>
            <Cell className="mdl-color--white" span={12}>
              <div>
                <TableComponent
                  title={title}
                  headers={headers}
                  items={items}
                  selectedItem={-1}
                  onSelect={() => {}}
                />
              </div>
            </Cell>
          </Inner>
        </Grid>
      );
    } else if (active === 3) {
      const { params } = this.props.admin;

      const emailServer = this.state.emailParams || params.emailServer || {};
      const backend = this.state.backendParams || params.backend || {};
      // const tunnelParams = this.state.tunnelParams || backend.tunnel || {};
      /* const hasTunnelParams = !!this.state.tunnelParams; */
      const saveBackendDisabled = !(
        this.state.backendParams || this.state.tunnelParams
      );
      const saveEmailDisabled = !this.state.emailServerParams;
      content = (
        <Grid>
          <Inner>
            <Cell className="mdl-color--white" span={12}>
              <div style={infoStyleD}>
                Backend configuration
                <Button
                  raised
                  disabled={saveBackendDisabled}
                  style={{ float: "right" }}
                  onClick={(e) => {
                    e.preventDefault();
                    this.onSaveBackend();
                  }}
                >
                  SAVE
                </Button>
              </div>
              <form style={infoStyleD}>
                <div style={{ width: "520px" }}>
                  <TextField
                    onChange={() => {}}
                    label="Public Api url"
                    style={{ width: FORM_WIDTH }}
                    value={backend.publicUrl}
                  />
                  <Icon
                    /* colored={hasTunnelParams} */
                    style={{ float: "right", marginTop: "8px" }}
                    name="link"
                    onClick={(e) => {
                      e.preventDefault();
                      this.displayTunnelDialog();
                    }}
                  />
                </div>
                <div>
                  <TextField
                    onChange={() => {}}
                    label="Api url"
                    disabled
                    style={{ width: FORM_WIDTH }}
                    value={backend.apiUrl}
                  />
                </div>
                <div>
                  <TextField
                    onChange={() => {}}
                    label="Auth url"
                    disabled
                    style={{ width: FORM_WIDTH }}
                    value={backend.authUrl}
                  />
                </div>
                <div>
                  <TextField
                    onChange={() => {}}
                    label="AppId"
                    disabled
                    style={{ width: FORM_WIDTH }}
                    value={backend.clientId}
                  />
                </div>
                <div>
                  <TextField
                    onChange={() => {}}
                    label="Secret"
                    disabled
                    style={{ width: FORM_WIDTH }}
                    value={backend.clientSecret}
                  />
                </div>
              </form>
              <div />
            </Cell>
          </Inner>
          <Inner>
            <Cell className="mdl-color--white" span={12}>
              <div style={infoStyleD}>
                Email server configuration
                <Button
                  raised
                  disabled={saveEmailDisabled}
                  style={{ float: "right" }}
                >
                  SAVE
                </Button>
              </div>
              <form style={infoStyleD} autoComplete="nope">
                <div>
                  <TextField
                    onChange={() => {}}
                    label="Server address"
                    style={{ width: FORM_WIDTH }}
                    value={emailServer.url}
                  />
                </div>
                <div>
                  <TextField
                    onChange={() => {}}
                    label="Username"
                    autoComplete="new-password"
                    style={{ width: FORM_WIDTH }}
                    value={emailServer.username}
                  />
                </div>
                <div>
                  <TextField
                    onChange={() => {}}
                    label="Password"
                    autoComplete="new-password"
                    type="password"
                    style={{ width: FORM_WIDTH }}
                    value={emailServer.password}
                  />
                </div>
              </form>
              <div />
            </Cell>
          </Inner>
          <Inner>
            <Cell className="mdl-color--white" span={12}>
              <div style={infoStyleD}>
                <span style={{ color: "#d50000" }}>Delete this assistant</span>
                <Button
                  raised
                  style={{
                    float: "right",
                    marginBottom: "24px",
                    backgroundColor: "#d50000",
                  }}
                >
                  DELETE
                </Button>
              </div>
              <div />
            </Cell>
          </Inner>
        </Grid>
      );
    }
    return (
      <div className="mdl-layout__content mdl-color--grey-100">
        <section>{content}</section>
      </div>
    );
  }
}

AdminManager.defaultProps = {
  activeTab: 0,
  admin: null,
  bot: null,
  isLoading: false,
  isSignedIn: false,
  profile: {},
  timezones,
  user: null,
};

AdminManager.propTypes = {
  activeTab: PropTypes.number,
  admin: PropTypes.shape({ params: PropTypes.shape({}).isRequired }),
  apiSaveBotRequest: PropTypes.func.isRequired,
  apiSetAdminParametersRequest: PropTypes.func.isRequired,
  appSetTitle: PropTypes.func.isRequired,
  bot: PropTypes.shape({
    name: PropTypes.string.isRequired,
    description: PropTypes.string,
    language: PropTypes.string,
    timezone: PropTypes.string,
  }),
  isLoading: PropTypes.bool,
  isSignedIn: PropTypes.bool,
  profile: PropTypes.shape({}),
  timezones: PropTypes.arrayOf(PropTypes.string).isRequired,
  user: PropTypes.shape({}),
};

const mapStateToProps = (state) => {
  const { admin } = state.app;
  const isSignedIn = state.user ? state.user.isSignedIn : false;
  const isLoading = state.loading;
  const selectedBotId = state.app ? state.app.selectedBotId : null;
  const { user } = state;
  const profile = user ? user.profile : null;

  // TODO get selectedBot from selectBotId
  const bot = selectedBotId ? admin.bots[0] : null;

  return {
    admin,
    isLoading,
    isSignedIn,
    selectedBotId,
    bot,
    user,
    profile,
  };
};

const mapDispatchToProps = (dispatch) => ({
  appSetTitle: (titleName) => {
    dispatch(appSetTitle(titleName));
  },
  apiSaveBotRequest: (params) => {
    dispatch(apiSaveBotRequest(params));
  },
  apiSetAdminParametersRequest: (params) => {
    dispatch(apiSetAdminParametersRequest(params));
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(AdminManager);
