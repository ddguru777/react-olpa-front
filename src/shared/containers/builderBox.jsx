import React, { Component } from "react";
import PropTypes from "prop-types";
import { Button } from "react-mdl";
import { DialogManager } from "zoapp-ui";
import { connect } from "react-redux";
import PublishDialog from "./publishDialog";

class BuilderBox extends Component {
  constructor(props) {
    super(props);
    this.state = { };
  }

  onAction = (action) => {
    console.log("TODO Publish");
    if (action === "Publish") {
      return false;
    }
    return true;
  }

  handleOpenPublishDialog = () => {
    const dialog = <PublishDialog open onAction={this.onAction} store={this.props.store} />;
    DialogManager.open({ dialog });
  }

  render() {
    if (this.props.isSignedIn) {
      const style = { paddingRight: "20%" };
      return (
        <div style={style}>
          <Button
            raised
            colored
            onClick={(e) => { e.preventDefault(); this.handleOpenPublishDialog(); }}
          >
          Publish
          </Button>
        </div>);
    }
    return (<div />);
  }
}

BuilderBox.defaultProps = {
  isSignedIn: false,
  store: null,
};

BuilderBox.propTypes = {
  isSignedIn: PropTypes.bool,
  store: PropTypes.shape({}),
};

const mapStateToProps = (state) => {
  const selectedBotId = state.app ? state.app.selectedBotId : null;
  const isSignedIn = state.user ? state.user.isSignedIn : false;
  const isLoading = state.app.loading;
  return { selectedBotId, isSignedIn, isLoading };
};

export default connect(mapStateToProps)(BuilderBox);
