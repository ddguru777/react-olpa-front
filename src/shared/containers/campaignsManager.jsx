import React, { Component } from "react";
import PropTypes from "prop-types";
import { Content } from "react-mdl";
import { connect } from "react-redux";
import { appSetTitle } from "../actions/app";


class CampaignsManager extends Component {
  componentWillMount() {
    this.props.appSetTitle("Campaigns");
  }

  render() {
    return (
      <Content className="mdl-color--grey-100">
        <section className="text-section" style={{ margin: "40px" }}>
          <div>Campaigns : TODO</div>
        </section>
      </Content>
    );
  }
}

CampaignsManager.propTypes = {
  appSetTitle: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => {
  const { admin } = state.app;
  const isSignedIn = state.user ? state.user.isSignedIn : false;
  const isLoading = state.loading;
  return { admin, isLoading, isSignedIn };
};

const mapDispatchToProps = dispatch => ({
  appSetTitle: (titleName) => {
    dispatch(appSetTitle(titleName));
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(CampaignsManager);
