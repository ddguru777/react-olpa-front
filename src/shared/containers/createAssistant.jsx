import React, { Component } from "react";
import PropTypes from "prop-types";
import { Content, Button, Textfield } from "react-mdl";
import { DialogManager, Selectfield } from "zoapp-ui";
import { connect } from "react-redux";
import { withRouter } from "react-router";

import TemplatesList from "../components/templatesList";
import ProcessingDialog from "./processingDialog";
import { apiCreateBot } from "../actions/api";
import { appSetTitle } from "../actions/app";

const formStyle = {
  paddingLeft: "16px",
};

const advancedStyle = {
  marginLeft: "16px",
  color: "rgba(0, 0, 0, 0.54)",
};

const boxStyle = {
  margin: "16px",
};

const headerStyle = {
  padding: "16px",
};

const h4 = {
  marginTop: "0px",
  marginBottom: "16px",
  fontSize: "16px",
  color: "rgba(0, 0, 0, 0.87)",
};
const secText = {
  color: "rgba(0, 0, 0, 0.54)",
};

// TODO get templates from API
const templates = [
  { id: 1, name: "Empty" },
  { id: 2, name: "HelloWorld" },
  { id: 3, name: "Test1" },
  { id: 4, name: "Import" },
];

class CreateAssistant extends Component {
  constructor(props) {
    super(props);
    this.state = { selectedTemplate: 0, template: null, loading: false };
  }

  componentWillMount() {
    this.props.appSetTitle("Create your virtual assistant");
  }

  componentDidUpdate() {
    if (this.state.loading && this.props.isLoading === false) {
      this.handleCloseCreateDialog();
    }
  }

  onImportTemplate = (data) => {
    // TODO check if json
    const json = JSON.parse(data);
    this.onSelectTemplate(3, json);
  }

  onSelectTemplate = (selected, data) => {
    let template = data;
    if (!template) {
      template = templates[selected];
    }
    this.setState({ selectedTemplate: selected, template });
  }

  handleCloseCreateDialog = () => {
    this.setState({ loading: false });
    DialogManager.close();
    if (!this.props.error) {
      this.props.history.push("/builder");
    }
  }

  handleCreate = () => {
    // Check textfields before processing
    const name = this.nameField.inputRef.value;
    const email = this.emailField.inputRef.value;
    const username = this.usernameField.inputRef.value;
    const password = this.passwordField.inputRef.value;
    const language = this.selectField.inputRef.value || "en";
    if (this.state.loading === false && name !== "" && email !== "" && username !== "" && password !== "") {
      const { template } = this.state;
      const botParams = {
        name, email, username, password, template, language,
      };
      this.setState({ loading: true });
      const dialog = <ProcessingDialog open onClosed={this.handleCloseDialog} />;
      DialogManager.open({ dialog });
      this.props.createBot(botParams);
    } else {
      // TODO display errors in dialogs
    }
  }

  handleLanguageChange = () => {
    const language = this.selectField.inputRef.value;
    console.log("handleCreate language=", language);
  }

  render() {
    const selected = this.state.selectedTemplate;
    // TODO json only for instance
    const acceptImport = "application/json";
    return (
      <Content className="mdl-color--grey-100">
        <section className="mdl-color--white mdl-shadow--2dp" style={boxStyle}>
          <div style={headerStyle}>
            <h4 style={h4}>Opla !</h4>
            <div style={secText}>
              Welcome to our five minutes installation process! Just fill in
              the informations below and you will get the most powerfull
              and extendable bot platform in the world.
            </div>
          </div>
        </section>
        <section className="mdl-color--white mdl-shadow--2dp" style={boxStyle}>
          <div style={headerStyle}>
            <h4 style={h4}>Templates</h4>
            <div style={secText}>
              Choose a prebuild asssistant, import one  or select an empty model.
            </div>
          </div>
          <TemplatesList
            items={templates}
            selectedItem={selected}
            onSelect={this.onSelectTemplate}
            onImport={this.onImportTemplate}
            acceptImport={acceptImport}
          />
          <div style={headerStyle}>
            <div style={secText}>
              Want more ?
              <br />
              In a near future we will release our BotStore to find
              the perfect bot from our community.
              <br />
            </div>
          </div>
        </section>
        <section className="mdl-color--white mdl-shadow--2dp" style={boxStyle}>
          <div style={headerStyle}>
            <h4 style={h4}>Informations needed</h4>
            <div style={secText}>
              Please provide the following information.
              Don&apos;t worry. You can always change them later.
            </div>
          </div>
          <form style={formStyle} autoComplete="new-password">
            <div><Textfield
              onChange={this.handleNameChange}
              label="Assistant name"
              floatingLabel
              style={{ maxWidth: "400px" }}
              ref={(input) => { this.nameField = input; }}
            />
            </div>
            <div><Textfield
              onChange={this.handleEmailChange}
              label="Username"
              floatingLabel
              autoComplete="new-password"
              style={{ maxWidth: "400px" }}
              ref={(input) => { this.usernameField = input; }}
            />
            </div>
            <div><Textfield
              onChange={this.handleEmailChange}
              label="Password"
              floatingLabel
              type="password"
              autoComplete="new-password"
              style={{ maxWidth: "400px" }}
              ref={(input) => { this.passwordField = input; }}
            />
            </div>
            <div><Textfield
              onChange={this.handleEmailChange}
              label="Your email"
              floatingLabel
              style={{ maxWidth: "400px" }}
              ref={(input) => { this.emailField = input; }}
            />
            </div>
            <div>
              <Selectfield
                label="Choose language"
                onChange={this.handleLanguageChange}
                floatingLabel
                ref={(input) => { this.selectField = input; }}
              >
                <option value="en">English</option>
                <option value="fr">French</option>
              </Selectfield>
            </div>
          </form>
        </section>
        <section style={boxStyle}>
          <Button
            raised
            accent
            ripple
            onClick={(e) => { e.preventDefault(); this.handleCreate(); }}
          >
          Let&apos;s go
          </Button>
          <Button ripple style={advancedStyle}>Advanced settings</Button>
        </section>
      </Content>
    );
  }
}

CreateAssistant.defaultProps = {
  error: null,
};

CreateAssistant.propTypes = {
  isLoading: PropTypes.bool.isRequired,
  error: PropTypes.string,
  createBot: PropTypes.func.isRequired,
  appSetTitle: PropTypes.func.isRequired,
  history: PropTypes.shape({ length: PropTypes.number, push: PropTypes.func }).isRequired,
};

const mapStateToProps = (state) => {
  const { admin, error } = state.app;
  const isSignedIn = state.user ? state.user.isSignedIn : false;
  const isLoading = state.app.loading;
  return {
    admin, isLoading, isSignedIn, error,
  };
};

const mapDispatchToProps = dispatch => ({
  createBot: (botParams) => {
    dispatch(apiCreateBot(botParams));
  },
  appSetTitle: (titleName) => {
    dispatch(appSetTitle(titleName));
  },
});

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(CreateAssistant));
