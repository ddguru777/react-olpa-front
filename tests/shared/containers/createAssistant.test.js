/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the GPL v2.0+ license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React from "react";
import renderer from "react-test-renderer";

import { CreateAssistantBase } from "shared/containers/createAssistant";

describe("containers/CreateAssistant", () => {
  it("renders correctly", () => {
    const appSetTitleSpy = jest.fn();
    const createBotSpy = jest.fn();
    const historySpy = { length: 0, push: jest.fn() };

    const component = renderer.create(
      <CreateAssistantBase
        isLoading={false}
        createBot={createBotSpy}
        appSetTitle={appSetTitleSpy}
        history={historySpy}
      />,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    expect(appSetTitleSpy).toHaveBeenCalled();
    expect(createBotSpy).not.toHaveBeenCalled();
    expect(historySpy.push).not.toHaveBeenCalled();
  });
});
