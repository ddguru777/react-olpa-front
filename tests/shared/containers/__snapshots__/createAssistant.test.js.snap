// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/CreateAssistant renders correctly 1`] = `
<div
  className="mdl-layout__content mdl-color--grey-100"
>
  <section
    className="mdl-color--white mdl-shadow--2dp"
    style={
      Object {
        "margin": "16px",
      }
    }
  >
    <div
      style={
        Object {
          "padding": "16px",
        }
      }
    >
      <h4
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontSize": "16px",
            "marginBottom": "16px",
            "marginTop": "0px",
          }
        }
      >
        Opla !
      </h4>
      <div
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.54)",
          }
        }
      >
        Welcome to our five minutes installation process! Just fill in the informations below and you will get the most powerfull and extendable bot platform in the world.
      </div>
    </div>
  </section>
  <section
    className="mdl-color--white mdl-shadow--2dp"
    style={
      Object {
        "margin": "16px",
      }
    }
  >
    <div
      style={
        Object {
          "padding": "16px",
        }
      }
    >
      <h4
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontSize": "16px",
            "marginBottom": "16px",
            "marginTop": "0px",
          }
        }
      >
        Templates
      </h4>
      <div
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.54)",
          }
        }
      >
        Choose a prebuild asssistant, import one or select an empty model.
      </div>
    </div>
    <div
      className="mdc-layout-grid"
    >
      <div
        className="mdc-layout-grid__inner"
      >
        <div
          className="mdl-shadow--2dp selectableListItem mdc-layout-grid__cell mdc-layout-grid__cell--span-2"
          style={
            Object {
              "height": "168px",
            }
          }
        >
          <div
            onClick={[Function]}
            role="presentation"
            style={
              Object {
                "textDecoration": "none",
                "width": "100%",
              }
            }
          >
            <div
              style={
                Object {
                  "color": "#666",
                  "fontSize": "16px",
                  "fontWeight": "400",
                  "height": "87%",
                  "lineHeight": "1.1",
                  "padding": "16px",
                }
              }
            >
              <div
                style={
                  Object {
                    "display": "table-row",
                    "textAlign": "center",
                    "width": "100%",
                  }
                }
              >
                Empty
              </div>
              <div
                style={
                  Object {
                    "display": "table-row",
                    "textAlign": "center",
                    "width": "100%",
                  }
                }
              >
                <img
                  alt="Empty"
                  src="./images/robots/robot-0.svg"
                  style={
                    Object {
                      "margin": "30px",
                      "width": "40%",
                    }
                  }
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="mdl-shadow--2dp selectableListItem mdc-layout-grid__cell mdc-layout-grid__cell--span-2"
          style={
            Object {
              "height": "168px",
            }
          }
        >
          <div
            onClick={[Function]}
            role="presentation"
            style={
              Object {
                "textDecoration": "none",
                "width": "100%",
              }
            }
          >
            <div
              style={
                Object {
                  "color": "#666",
                  "fontSize": "16px",
                  "fontWeight": "400",
                  "height": "87%",
                  "lineHeight": "1.1",
                  "padding": "16px",
                }
              }
            >
              <div
                style={
                  Object {
                    "display": "table-row",
                    "textAlign": "center",
                    "width": "100%",
                  }
                }
              >
                Import
              </div>
              <div
                style={
                  Object {
                    "display": "table-row",
                    "textAlign": "center",
                    "width": "100%",
                  }
                }
              >
                <form
                  style={
                    Object {
                      "width": "100%",
                    }
                  }
                >
                  <div
                    className="fileInput"
                  >
                    <i
                      className="material-icons"
                    >
                      cloud_upload
                    </i>
                    <input
                      accept="application/json"
                      onChange={[Function]}
                      type="file"
                    />
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "padding": "16px",
        }
      }
    >
      <div
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.54)",
          }
        }
      >
        Want more ?
        <br />
        In a near future we will release our BotStore to find the perfect bot from our community.
        <br />
      </div>
    </div>
  </section>
  <form
    autoComplete="new-password"
    id="create-assistant-form"
    onSubmit={[Function]}
  >
    <section
      className="mdl-color--white mdl-shadow--2dp"
      style={
        Object {
          "margin": "16px",
        }
      }
    >
      <div
        style={
          Object {
            "padding": "16px",
          }
        }
      >
        <h4
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontSize": "16px",
              "marginBottom": "16px",
              "marginTop": "0px",
            }
          }
        >
          Informations needed
        </h4>
        <div
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.54)",
            }
          }
        >
          Please provide the following information. Don't worry. You can always change them later.
        </div>
        <div>
          <div
            className="mdc-text-field mdc-text-field--upgraded"
            style={
              Object {
                "width": "400px",
              }
            }
          >
            <input
              className="mdc-text-field__input"
              defaultValue=""
              id="create-assistant-name"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
            />
            <label
              className="mdc-floating-label"
              focused="false"
              htmlFor="create-assistant-name"
              style={Object {}}
            >
              Assistant name
            </label>
            <div
              className="mdc-text-field__bottom-line"
            />
          </div>
        </div>
        <div>
          <div
            className="mdc-text-field mdc-text-field--upgraded"
            style={
              Object {
                "width": "400px",
              }
            }
          >
            <input
              autoComplete="new-password"
              className="mdc-text-field__input"
              defaultValue=""
              id="create-assistant-username"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
            />
            <label
              className="mdc-floating-label"
              focused="false"
              htmlFor="create-assistant-username"
              style={Object {}}
            >
              Username
            </label>
            <div
              className="mdc-text-field__bottom-line"
            />
          </div>
        </div>
        <div>
          <div
            className="mdc-text-field mdc-text-field--upgraded"
            style={
              Object {
                "width": "400px",
              }
            }
          >
            <input
              autoComplete="new-password"
              className="mdc-text-field__input"
              defaultValue=""
              id="create-assistant-password"
              minLength="4"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="password"
            />
            <label
              className="mdc-floating-label"
              focused="false"
              htmlFor="create-assistant-password"
              style={Object {}}
            >
              Password (min. 4 characters)
            </label>
            <div
              className="mdc-text-field__bottom-line"
            />
          </div>
        </div>
        <div>
          <div
            className="mdc-text-field mdc-text-field--upgraded"
            style={
              Object {
                "width": "400px",
              }
            }
          >
            <input
              className="mdc-text-field__input"
              defaultValue=""
              id="create-assistant-email"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="email"
            />
            <label
              className="mdc-floating-label"
              focused="false"
              htmlFor="create-assistant-email"
              style={Object {}}
            >
              Your email
            </label>
            <div
              className="mdc-text-field__bottom-line"
            />
          </div>
        </div>
        <div>
          <div
            aria-hidden={true}
            className="mdc-select"
            onClick={[Function]}
            onKeyUp={[Function]}
            role="listbox"
            style={
              Object {
                "width": "400px",
              }
            }
          >
            <div
              className="mdc-select__surface"
              tabIndex="0"
            >
              <div
                className="mdc-select__label mdc-select__label--float-above"
              >
                Choose language
              </div>
              <div
                className="mdc-select__selected-text"
              >
                English
              </div>
              <div
                className="mdc-select__bottom-line"
              />
            </div>
            <div
              className="mdc-select__menu mdc-menu"
              tabIndex="-1"
            >
              <ul
                aria-hidden="true"
                className="mdc-menu__items mdc-list"
                role="menu"
              >
                <li
                  aria-selected="true"
                  className="mdc-list-item"
                  onClick={[Function]}
                  onKeyUp={[Function]}
                  role="option"
                  tabIndex="0"
                >
                  English
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </section>
    <section
      style={
        Object {
          "margin": "16px",
        }
      }
    >
      <button
        className="mdc-button mdc-button--raised"
        disabled={false}
        id={null}
        onClick={null}
        type="submit"
      >
        
        Let's go
      </button>
      <button
        className="mdc-button"
        disabled={false}
        id={null}
        onClick={null}
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "marginLeft": "16px",
          }
        }
        type="button"
      >
        
        Advanced settings
      </button>
    </section>
  </form>
</div>
`;
